### PR TITLE
Update installed JDKs to match current boot JDK requirements

### PR DESCRIPTION
**Pull request by sxa**
##### Checklist

- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [x] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access): https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/

Draft pending VPC checks and verification that the build uses the installed JDK20 but approvals welcome in the meantime.

### CODE REVIEW

**Comment by sxa** [comment]
This is going to require a bit more work because it's trying to pull JDK20 from api.adoptopenjdk.net which does redirect to adoptium, so I'm going to stick this in draft for now until it can be progressed.

**Comment by sxa** [comment]
On hold until we come up with something sane to do for Linux on s390x and arm32

**Comment by karianna** [comment]
@sxa Needs a rebase

**Comment by sxa** [comment]
VPC: https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/1845/

**Review by adamfarley — CHANGES_REQUESTED** [review]
Most of it LGTM. 2 concerns:

- Various Yaml Lint and Ansible lint failures. The latter seem to be caused by the "---" being before the first comment block in a few files, as the error doesn't seem to pop up when the "---" is after the first comment block.
- Lines 190 and 223 in ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk_install/tasks/main.yml look like a merge has gone wrong. Pls fix thnx. :)

**Comment by sxa** [comment]
GPG signature verification is failing with this change so some extra remediation will be required for that.

**Comment by sxa** [comment]
AIX and Linux (including riscv64 tested separately as it has different paths through the playbooks) are working properly now. Windows still undergoing testing...

**Comment by sxa** [comment]
Windows now looking good: https://ci.adoptium.net/job/VagrantPlaybookCheck/1855/

**Comment by sxa** [comment]
CentOS6 and Solaris also now clean (Both have special paths in this particular role at present.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk_install/tasks/main.yml [hunk 1]
```diff
@@ -1,12 +1,28 @@
 ---
-##############################################################
-# ansible install adoptopenjdk into /usr/lib/jvm/JDK_VERSION #
-##############################################################
+###############################################################
+# ansible install adopt binary  into /usr/lib/jvm/JDK_VERSION #
```
Comment (adamfarley) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk_install/tasks/main.yml:3]:
```suggestion
# ansible install adopt binary into /usr/lib/jvm/JDK_VERSION #
```

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk_install/tasks/main.yml [hunk 2]
```diff
@@ -1,12 +1,28 @@
 ---
-##############################################################
-# ansible install adoptopenjdk into /usr/lib/jvm/JDK_VERSION #
-##############################################################
+###############################################################
```
Comment (adamfarley) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk_install/tasks/main.yml:2]:
```suggestion
##############################################################
```

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk_install/tasks/main.yml [hunk 3]
```diff
@@ -1,12 +1,28 @@
...
+# ansible install adopt binary into /usr/lib/jvm/JDK_VERSION #
+###############################################################
```
Comment (adamfarley) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk_install/tasks/main.yml:4]:
```suggestion
##############################################################
```

Comment (adamfarley) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/adoptopenjdk_install/tasks/main.yml:4]: Lining up comment hashes.

**Review by adamfarley — APPROVED** [review]
LGTM

**Comment by adamfarley** [comment]
And thanks for handling all those linter changes. :)

**Review by steelhead31 — APPROVED** [review]
LGTM

**Comment by sxa** [comment]
Merging so that it's ready for the JDK22 dry run pipelines later this week.

**Comment by sxa** [comment]
Rebuild of linux images happening under https://ci.adoptium.net/job/centos7_docker_image_updater/471

**Pull request merged**